### PR TITLE
Use UTF-8 I/O encoding for Python 3

### PR DESCRIPTION
**Pull request by compnerd**
Replace this paragraph with a description of your changes and rationale. Provide links to external references/discussions if appropriate.

Resolves SR-NNNN.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @drexin @tbkka @tachoknight

**Comment by compnerd** [comment]
@swift-ci please test

File: utils/gyb.py [hunk 1]
```diff
@@ -1208,12 +1209,12 @@ def succ(a):
         help='''Bindings to be set in the template's execution context''')
 
     parser.add_argument(
-        'file', type=argparse.FileType(),
+        'file', type=str,
         help='Path to GYB template file (defaults to stdin)', nargs='?',
         default=sys.stdin)
```
Comment (tbkka) [inline utils/gyb.py:1214]: Do you need to update the default here?

**Review by tbkka — APPROVED** [review]

File: utils/gyb.py (hunk 1 above)
Comment (compnerd) [reply utils/gyb.py:1214]: Ah, so I do.  Good catch

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by tbkka** [comment]
Ugh. 
```
Traceback (most recent call last):
  File "/Volumes/swift-ci/jenkins/workspace/swift-PR-osx/branch-master/swift/utils/gyb.py", line 1269, in <module>
    main()
  File "/Volumes/swift-ci/jenkins/workspace/swift-PR-osx/branch-master/swift/utils/gyb.py", line 1253, in main
    ast = parse_template(args.file, f.read())
  File "/Volumes/swift-ci/jenkins/workspace/swift-PR-osx/branch-master/swift/utils/gyb.py", line 1064, in parse_template
    return Block(ParseContext(filename, text))
  File "/Volumes/swift-ci/jenkins/workspace/swift-PR-osx/branch-master/swift/utils/gyb.py", line 635, in __init__
    self.children.append(node(context))
  File "/Volumes/swift-ci/jenkins/workspace/swift-PR-osx/branch-master/swift/utils/gyb.py", line 654, in __init__
    context.next_token()
  File "/Volumes/swift-ci/jenkins/workspace/swift-PR-osx/branch-master/swift/utils/gyb.py", line 553, in next_token
    for kind in self.tokens:
  File "/Volumes/swift-ci/jenkins/workspace/swift-PR-osx/branch-master/swift/utils/gyb.py", line 502, in token_generator
    self.template, code_start, self.line_starts)
  File "/Volumes/swift-ci/jenkins/workspace/swift-PR-osx/branch-master/swift/utils/gyb.py", line 142, in tokenize_python_to_unmatched_close_curly
    stream = StringIO(source_text)
UnicodeEncodeError: 'ascii' codec can't encode character u'\u25bf' in position 1800: ordinal not in range(128)
```

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
@swift-ci please test macOS platform

**Comment by compnerd** [comment]
@swift-ci please test macOS platform

**Comment by compnerd** [comment]
@swift-ci please test Windows platform

**Pull request merged**